src/consumer/subscribed.rs

```diff
@@ -287,73 +342,107 @@ where
     fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
         loop {
             self.state = match self.state {
-                State::Joining(ref mut join_group) => {
-                    try_ready!(join_group.poll());
-
-                    if let Some(ref coordinator) = self.coordinator {
-                        State::UpdatingOffsets(coordinator.update_offsets())
-                    } else {
-                        State::updating(self.subscriptions.clone(), self.fetcher.clone())
+                State::Rebalancing(ref coordinator, ref mut joining) => match joining.poll() {
+                    Ok(Async::Ready(_)) => State::Updating(coordinator.update_offsets()),
+                    Ok(Async::NotReady) => {
+                        if let Some((record, _)) = self.records.pop_back() {
```

Comment:
why pop_back(), wouldn't it retrieve records in reverse order?

Reply:
oops, I missed it